plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/BaseKtlintCheckTask.kt

```diff
@@ -104,39 +105,54 @@ abstract class BaseKtlintCheckTask(
         project.javaexec(generateJavaExecSpec(additionalConfig()))
     }
 
-    abstract fun additionalConfig(): (JavaExecSpec) -> Unit
+    @OutputFiles
+    private fun ktlintArgsFile() = newFileProperty(objectFactory, projectLayout).apply {
+        set(
+            project.layout.buildDirectory.file(
+                project.provider {
+                    "ktlint/${this@BaseKtlintCheckTask.name}.args"
+                }
+            )
+        )
+    }
```

Comment:
Shouldn't this be a lazy prop instead of a method? Currently every time you execute this method, a new property will be generated.